src/test/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperatorTest.scala

```diff
@@ -3,210 +3,64 @@ package tech.cryptonomic.conseil.tezos
-  "getBlocks" should "work correctly with an offset and hint" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 3, Some("BKiRLq7c2QVr6X428RRvp6JLTJEnWPE4bc4cAQHoo9GuZz9GH38"), followFork = false)
-    blocks.get.head.metadata.hash should be ("BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe")
-  }
+    val nodeOp: TezosNodeOperator = new TezosNodeOperator(tezosRPCInterface, config)
 
+    //when
+    val block: Future[TezosTypes.Block] = nodeOp.getBlock("zeronet", BlockHash("BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe"))
 
-  "getBlocks" should "work correctly with an offset" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 1, None, followFork = false)
-    blocks.get.size should be (1)
+    //then
+    Await.result(block, 1.second).metadata.hash shouldBe BlockHash("BLJKK4VRwZk7qzw64NfErGv69X4iWngdzfBABULks3Nd33grU6c")
   }
 
+  "getAllBlocks" should "should correctly fetch all the blocks" in {
+    //given
+    (tezosRPCInterface.runAsyncGetQuery _).when("zeronet", "blocks/head~").returns(Future.successful(TezosResponseBuilder.blockResponse))
+    (tezosRPCInterface.runAsyncGetQuery _).when("zeronet", "blocks/head/operations").returns(Future.successful(TezosResponseBuilder.operationsResponse))
 
-  // Once we can mock the datbase, we should test whether getBlocks() works on a forked chain.
+    (tezosRPCInterface.runBatchedGetQuery[Int] _).when("zeronet", (0 to 162099).toList, *, *).returns(Future.successful(List((0, TezosResponseBuilder.batchedGetQueryResponse))))
+    (tezosRPCInterface.runBatchedGetQuery[BlockHash] _).when("zeronet", *, *, *).returns(Future.successful(List((BlockHash("BMKoXSqeytk6NU3pdL7q8GLN8TT7kcodU1T6AUxeiGqz2gffmEF"), TezosResponseBuilder.batchedGetQuerySecondCallResponse))))
 
-  "getBlocks" should "work correctly with a hint whose level is too high" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 0, 1, Some("BKiiqiGu758Q76DLiqvN2ocwowjnR3aRrXguRYVq2xw61chzQoZ"), followFork = false)
-    blocks.get.length should be (2)
-  }
+    val nodeOp: TezosNodeOperator = new TezosNodeOperator(tezosRPCInterface, config)
 
+    //when
+    val block: Future[List[(TezosTypes.Block, List[TezosTypes.AccountId])]] = nodeOp.getAllBlocks("zeronet")
 
-  "getAccounts" should "correctly fetch all accounts for a block" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val accounts = nodeOp.getAccounts("zeronet", "BKiRLq7c2QVr6X428RRvp6JLTJEnWPE4bc4cAQHoo9GuZz9GH38")
-    accounts.get.accounts.size should be (1) //actually 5 in tezos, 1 for testing purposes.
-    val account = accounts.get.accounts.get("tz1btz5Av9BdpoTPnS9zGyPvpgAovmaZ23iN")
-    account.get.balance should be (12000000000000.0)
+    //then
+    Await.result(block, 1.second) should have length 1
```

Comment:
Yeah, it would be just a dump response from the stub. I personally don't like these tests because it should be an integration test. I created a task for it to make it more reliable: https://github.com/Cryptonomic/Conseil/issues/222